src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -2430,5 +2431,28 @@ private bool ConsumeMultiLineComment(ReadOnlySpan<byte> localBuffer, int previou
             _tokenType = JsonTokenType.Comment;
             return true;
         }
+
+        private string DebuggerDisplay => $"TokenType = {DebugTokenType} (TokenStartIndex = {TokenStartIndex}) Consumed = {BytesConsumed}";
```

Comment:
CurrentDepth would be useful too.

Reply:
Based on the initial issue, should we have a few characters before/after of the current cursor, maybe with ellipses?

E.g. (assume `|` means the current reader position):
`{"foo": 1, |"bar": 2], {"baz": 3}`

The debugger display could show: `... 1, "bar":2]...`

Reply:
Seemed to get to be a bit long, thought it wasn't as useful in the reader.

Reply:
I was given feedback previously that knowing the json string in the vicinity of the reader could help callers debug why something went wrong (especially if the payload was invalid).